### PR TITLE
Image cell

### LINKED ISSUE #14: Cell for image

### PR DESCRIPTION
**Pull request by RoxaneBurri**
This PR:

- create cells for:
    - plain image survey
    - image and word-cloud survey
    - customer satisfaction (cast, nps and ueq)
    - demographic question
- manage mutation with the database

close #14

### CODE REVIEW

File: redwood.toml [hunk 1]
```diff
@@ -7,11 +7,11 @@
 
 [web]
   title = "Redwood App"
-  port = 8924
-  apiUrl = "/.redwood/functions" # you can customise graphql and dbauth urls individually too: see https://redwoodjs.com/docs/app-configuration-redwood-toml#api-paths
+  port = 8929
+  apiUrl = "/.netlify/functions"
```
Comment (spaenleh) [inline redwood.toml:11]: Why do we need this ?

File: web/src/Routes.tsx [hunk 2]
```diff
@@ -13,34 +13,22 @@ import ScaffoldLayout from 'src/layouts/ScaffoldLayout'
 
 import { useAuth } from './auth'
 import NavigationLayout from './layouts/NavigationLayout/NavigationLayout'
+import ForgotPasswordPage from './pages/ForgotPasswordPage/ForgotPasswordPage'
+import LoginPage from './pages/LoginPage/LoginPage'
+import SignupPage from './pages/SignupPage/SignupPage'
```
Comment (spaenleh) [inline web/src/Routes.tsx:18]: You don't need to import those. they are auto-imported by redwood. If you IDE does not pick that up, you should select the typescript version to use the one from the workspace.

File: web/src/Routes.tsx [hunk 3]
```diff
@@ -13,34 +13,22 @@ import ScaffoldLayout from 'src/layouts/ScaffoldLayout'
...
-      </Set>
-      <Set wrap={ScaffoldLayout} title="Users" titleTo="users" buttonLabel="New User" buttonTo="newUser">
-        <Route path="/users/new" page={UserNewUserPage} name="newUser" />
-        <Route path="/users/{id:Int}/edit" page={UserEditUserPage} name="editUser" />
-        <Route path="/users/{id:Int}" page={UserUserPage} name="user" />
-        <Route path="/users" page={UserUsersPage} name="users" />
-      </Set>
       <Set wrap={NavigationLayout}>
-        <Route path="/" page={HomePage} name="home" />
+        <Route path="/" page={HomePage} name="home" />
```
Comment (spaenleh) [inline web/src/Routes.tsx:27]: You should set the auto-format action on save with a plugin like prettier-eslint to remove the issues like these with trailing whitespaces.

File: web/src/Routes.tsx [hunk 4]
```diff
@@ -13,34 +13,22 @@ import ScaffoldLayout from 'src/layouts/ScaffoldLayout'
...
-        <Route path="/" page={HomePage} name="home" />
+        <Route path="/" page={HomePage} name="home" /> 
         <Private unauthenticated="signup">
           <Route path="/group-a" page={GroupAPage} name="groupA" />
           <Route path="/group-b" page={GroupBPage} name="groupB" />
-          <Route path="/group-b-global-question" page={GroupBGlobalQuestionPage} name="groupBGlobalQuestion" />
-          <Route path="/csat" page={CustomerSatPage} name="csat" />
-          <Route path="/nps" page={NetPromoterScorePage} name="nps" />
-          <Route path="/ueq" page={UserExperienceQuestionnairePage} name="ueq" />
+          <Route path="/customer-satisfaction" page={CustomerSatisfactionPage} name="customerSatisfaction" />
```
Comment (spaenleh) [inline web/src/Routes.tsx:31]: Should we rename this page to "/survey" ?

File: web/src/components/DemographicCell/DemographicCell.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+          text={['0-15 years old', '15-30 years old', '30-45 years old', '45-60 years old', '60+', 'Prefer not to say']}
+          direction="column"
+          value={demographic?.age.toString() || ''}
+          validation={{ required: true }}     
+        />
+        <LikertScaleQuestionField
+          name={DEMOGRAPHIC_EDUCATION}
+          n={6}
+          question="What is the highest degree or level of education you have completed?"
+          text={['CFC/Apprenticeship', 'Maturité/Baccalauérat', 'Bachelors Degree', 'Masters Degree', 'Ph.D or higher', 'Prefer not to say']}
```
Comment (spaenleh) [inline web/src/components/DemographicCell/DemographicCell.tsx:143]:
```suggestion
          text={['CFC/Apprenticeship', 'Maturité/Baccalauréat', 'Bachelors Degree', 'Masters Degree', 'Ph.D or higher', 'Prefer not to say']}
```

File: web/src/components/ImageCell/ImageCell.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+export const Failure = ({
+  error,
+}: CellFailureProps) => (
+  <div style={{ color: 'red' }}>Error: {error?.message}</div>
+)
+
+export const Success = ({
+  image,
+}: CellSuccessProps<FindPlainImageByImageNumber>) => {
+  const location = image.imageLocation
```
Comment (spaenleh) [inline web/src/components/ImageCell/ImageCell.tsx:28]: Do you need this ?

File: web/src/pages/CustomerSatisfactionPage/CustomerSatisfactionPage.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  const handleNextStep = () => {
+    setStep((s) => s + 1)
+    // navigate(routes.home(), {replace: true})
+  }
+  return (
+    <Container maxW="6xl">
+    <Stack direction="column" gap={8} alignItems="center">
+      <Text data-testid="instruction">
+        You are shown a picture and please answer some questions
+      </Text>
```
Comment (spaenleh) [inline web/src/pages/CustomerSatisfactionPage/CustomerSatisfactionPage.tsx:25]: Remove ?

File: redwood.toml (hunk 1 above)
Comment (RoxaneBurri) [reply redwood.toml:11]: I have an error with the port number with command `yarn rw dev`

File: web/src/Routes.tsx (hunk 2 above)
Comment (RoxaneBurri) [reply web/src/Routes.tsx:18]: Okay

File: web/src/Routes.tsx (hunk 3 above)
Comment (RoxaneBurri) [reply web/src/Routes.tsx:27]: I will add this

File: web/src/Routes.tsx (hunk 4 above)
Comment (RoxaneBurri) [reply web/src/Routes.tsx:31]: Yes as it is a customer satisfaction survey with demographic questions

File: web/src/components/ImageCell/ImageCell.tsx (hunk 6 above)
Comment (RoxaneBurri) [reply web/src/components/ImageCell/ImageCell.tsx:28]: Nop

File: web/src/pages/CustomerSatisfactionPage/CustomerSatisfactionPage.tsx (hunk 7 above)
Comment (RoxaneBurri) [reply web/src/pages/CustomerSatisfactionPage/CustomerSatisfactionPage.tsx:25]: We still need context and instruction on each page, but I need to update this

**Pull request merged**